tests/pubsub/testgossipinternal.nim

```diff
@@ -662,6 +662,67 @@ suite "GossipSub internal":
     await allFuturesThrowing(conns.mapIt(it.close()))
     await gossipSub.switch.stop()
 
+      # test cases for block 5 gossibsub test plan 
+      # check correctly parsed ihave/iwant/graft/prune/idontwant messages
+      # check value before & after decoding equal using protoc cmd tool for reference
+      # check encoding / decoding time less than 1 millisecond
+  asyncTest "Check RPCMsg encoding":
+    let backofftime = 10.uint64
+    var id: seq[byte] = @[123]
+    let rpcMsg = RPCMsg(
+      control: some(
+        ControlMessage(
+          ihave: @[ControlIHave(topicID: "foobar", messageIDs: @[id])],
+          iwant: @[ControlIWant(messageIDs: @[id])],
+          graft: @[ControlGraft(topicID: "foobar")],
+          prune: @[ControlPrune(topicID: "foobar", backoff: backofftime)],
+          idontwant: @[ControlIWant(messageIDs: @[id])],
+        )
+      )
+    )
+    let encodedExpected: seq[byte] =
+      @[
+        26, 51, 10, 10, 6, 102, 111, 111, 98, 97, 114, 18, 3, 49, 50, 51, 18, 5, 10, 3,
+        49, 50, 51, 26, 8, 10, 6, 102, 111, 111, 98, 97, 114, 34, 10, 10, 6, 102, 111,
+        111, 98, 97, 114, 16, 10, 42, 5, 10, 3, 49, 50, 51,
+      ] #encoded using protoc cmd tool
+
+    let encodeTimeout = Moment.now() + 1.milliseconds
```

Comment:
tests should never have timing-based components - this is testing whether the clock on the computer is working and the speed of the cpu, which is irrelevant for the given functionality

Reply:
timing check removed